tests/ReturnTypeTest.php

```diff
@@ -1208,7 +1208,7 @@ function foo(): array {
                       }
                       return $arr;
                     }',
-                'error_message' => 'LessSpecificReturnStatement',
+                'error_message' => 'InvalidReturnStatement',
```

Comment:
This is another issue where a `LessSpecific` issue doesn't work the way I would intuitively expect. Perhaps for non-sealed arrays this would make sense, `array{a: string, b: string}`-unsealed could also have `"c"` set, but Psalm resolves `$arr` to `non-empty-array<"a"|"b", "foo">`, which definitely does not have `"c"` set.

So we have types A = unsealed `array{a: string, b: string, c: string}` and B = `non-empty-array<"a"|"b", "foo">`. A cannot contained by B because B doesn't have `"c"` set. B cannot contain A because A has `"c"` set.

I think perhaps if `LessSpecific` issues are changed to work the way I would inuit them to, maybe we should add a `IncompleteReturnStatement` or something of that nature to indicate that A (return type) and B (return statement type) intersect, but neither contains the other?